twitter/View Controllers/TimelineViewController.m

```diff
@@ -60,4 +80,73 @@ - (IBAction)didTapLogout:(id)sender {
     
     [[APIManager shared] logout];
 }
+- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath {
+    Tweet *tweet = self.arrayOfTweets[indexPath.row];
+    TweetCell *tweetCell = [tableView dequeueReusableCellWithIdentifier:@"TweetCell" forIndexPath:indexPath];
+    tweetCell.tweet = tweet;
+    // set tweet profile picture
+    NSString *URLString = tweet.user.profilePicture;
+    NSURL *url = [NSURL URLWithString:URLString];
+    [tweetCell.profilePicture setImageWithURL:url];
+    
+    // set other metadata for tweet
+    tweetCell.screenName.text = tweet.user.name;
+    tweetCell.tweetText.text = tweet.text;
+    tweetCell.userName.text = [NSString stringWithFormat:@"@%@", tweet.user.screenName];
+    tweetCell.datePosted.text = tweet.createdAtString;
+    [tweetCell.replyButton setTitle:[NSString stringWithFormat:@"%d", tweet.replyCount] forState:UIControlStateNormal];
+    [tweetCell.favoriteButton setTitle:[NSString stringWithFormat:@"%d",tweet.favoriteCount] forState:UIControlStateNormal];
+    [tweetCell.retweetButton setTitle:[NSString stringWithFormat:@"%d", tweet.retweetCount] forState:UIControlStateNormal];
```

Comment:
You could refactor this code and have this logic wrapped inside a method in `TweetCell` called `updateWithTweet:`. This can help take the clutter out of the view controller and delegate to each view the process of setting it up with the model. Although one could claim this violates the principle of MVC... LOL